### PR TITLE
Hyperlinks

### PR DESCRIPTION
**Pull request by kahummer**
Should be merged after https://github.com/onaio/gisida-react/pull/218
Fixes: https://github.com/onaio/irfsomalia/issues/889,  https://github.com/onaio/irfsomalia/issues/894, https://github.com/onaio/irfsomalia/issues/897
This pr has changed the structure of our current menu to accommodate the hyperlinks. Some styling tweaks will be desired before we merge.
The configuration had to change to target sub-sector/ sub-category level navigation based on the parent sector as a way of discerning in the event of duplicate `sub-sectors`.
BRCis II is the parent sector.
e.g
```
"hyperLink": {
      "Boundaries": {
        "link": "https://drive.google.com/file/d/1W93Tw3MOhjTEz8h2ycUbY-k6yZXf81Pi/view",
      },
      "BRCiS II Output Indicators": {
        "link": "https://drive.google.com/file/d/1W93Tw3MOhjTEz8h2ycUbY-k6yZXf81Pi/view",
      }
```
More docs on this https://github.com/onaio/gisida-react/blob/hyperlinks/docs/features/menu-legend-links.md
The same specification will be used for the info Icon too. The legendInfo won't require this since this is built at the layer level.
Screenshots of the current implementation
Screenshot from 2020-08-03 20-09-02

- [x] Have consistent ui with previous implementations

- [x] Legend info and Hyperlink

- [x] Add tests

- [x] Add menu links documentation

- [x] Export unsets inner style

- [x] Handle subsectors inside subsectors

### CODE REVIEW

**Comment by kahummer** [comment]
Add links and info description to the legend
The following specification should be used
```
"legendHyperlink": "google.com",
"legendDescription": "Provision of emergency humanitarian assistance to vulnerable populations",
"legendHyperlinkText": "Google doc Report",
```
Screenshot from 2020-07-30 11-52-57

**Comment by kelvin-muchiri** [comment]
> The legendInfo won't require this since this is built at the layer level.

Instead of duplicating the description in the hyperlink, why can't we use the same info being used by the legend or vice versa? Or can one enter different descriptions for hyperlink and legend?

**Comment by kelvin-muchiri** [comment]
> Add links and info description to the legend
> The following specification should be used
> 
> ```
> "legendHyperlink": "google.com",
> "legendDescription": "Provision of emergency humanitarian assistance to vulnerable populations",
> "legendHyperlinkText": "Google doc Report",
> ```
> 
> Screenshot from 2020-07-30 11-52-57

Where is this information added? in the `hyperLink` object or?

File: docs/features/menu-legend-links.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+The `parent category` is used to identify the sub-sector based on parent sector/sub-sector. The parent category should be appended before sector/sub-sector key inside hyperlink property. This helps when we have same sub-sector names in different sectors.
+
+BRCis II is the parent in this case
+```
+"BRCiS II Output Indicators": {
+        "link": "https://gisida.com",
+        "description": "desc"
+      }
+```
+#### Legend (layer level) spec
```
Comment (kelvin-muchiri) [inline docs/features/menu-legend-links.md:34]: Could you please add a layer spec including these new properties so it's a bit clear where to add them in the layer spec?

File: src/lib/components/Filter/Filter.js [hunk 2]
```diff
@@ -283,15 +283,15 @@ export class Filter extends Component {
...
-    if (
-      nextProps.layerObj &&
-      nextProps.layerObj.aggregate.filter &&
-      !Object.keys(nextProps.FILTER).length
-    ) {
-      this.props.dispatch(
-        Actions.filtersUpdated(nextProps.mapId, nextProps.layerObj && nextProps.layerObj.id)
-      );
-    }
+    // if (
```
Comment (kelvin-muchiri) [inline src/lib/components/Filter/Filter.js]: Can we remove the commented code?

**Comment by kahummer** [comment]
@kelvin-muchiri Just figured we can use categories for the legend links. In that case, I'll be adding the link description only

**Pull request merged**